framework/src/interfaces/Coupleable.C

```diff
@@ -521,6 +521,20 @@ Coupleable::vectorTagValueHelper(const std::string & var_name,
   return vectorTagValueHelper<T>(var_name, tag, comp);
 }
 
+template <>
+const GenericVariableValue<false> &
+Coupleable::coupledGenericDofValue<false>(const std::string & var_name, unsigned int comp) const
+{
+  return coupledDofValues(var_name, comp);
+}
+
+template <>
+const GenericVariableValue<true> &
+Coupleable::coupledGenericDofValue<true>(const std::string & var_name, unsigned int comp) const
+{
+  return adCoupledValue(var_name, comp);
```

Comment:
This doesn't look right, or at least I don't think this would be what the user would expect. I think what you would like to be returning from this is an array where the value part of the dual number is the dof value and the derivative part of the dof value has a single value of one at the corresponding dof index. Is that right? If so, maybe I could work to help you implement that

Reply:
Oh wait... you already have this... so I think this just needs to be
```c++
  return adCoupledDofValues(var_name, comp);
```
Not sure why I'm unable to make a suggestion here :frowning:

Reply:
Oops!